omegaconf/_utils.py

```diff
@@ -240,7 +240,7 @@ def extract_dict_subclass_data(obj: Any, parent: Any) -> Optional[Dict[str, Any]
         warnings.warn(
             "Subclassing of `Dict` by Structured Config classes is deprecated",
             UserWarning,
-            stacklevel=2,
+            stacklevel=1,
         )
```

Comment:
With `stacklevel=2` we get a warning that looks like this:
```bash
$ python tmp.py
/home/jbss/omegaconf.git/omegaconf/_utils.py:373: UserWarning: Subclassing of `Dict` by Structured Config classes is deprecated
  dict_subclass_data = extract_dict_subclass_data(obj=obj, parent=dummy_parent)
```
With `stacklevel=1` we get a warning that looks like this:
```bash
$ python tmp.py
/home/jbss/omegaconf.git/omegaconf/_utils.py:240: UserWarning: Subclassing of `Dict` by Structured Config classes is deprecated
  warnings.warn(
```

Reply:
The exact interesting stack frame is probably not in a fixed depth.
Try to get it to issue the line calling `OmegaConf.create()` in the simple case.

Also, add the name of the type subclassing Dict to the warning.

Reply:
Done in https://github.com/omry/omegaconf/pull/707/commits/6eeb616e432eefa086fd17d7c813521417fc32de and https://github.com/omry/omegaconf/pull/707/commits/71ad2b61c27fca5aee0dce609f06a44e5e0ac096.